zerostash/src/commands/mount.rs

```diff
@@ -256,24 +244,102 @@ impl FilesystemMT for SimpleFs {
         size: u32,
         callback: impl FnOnce(ResultSlice<'_>) -> CallbackResult,
     ) -> CallbackResult {
-        println!("read: {:?} {:#x} @ {:#x}", path, size, offset);
-
-        let file_content = self.file_content.lock().unwrap();
+        debug!("read: {:?} {:#x} @ {:#x}", path, size, offset);
+        let size = size as usize;
+        let offset = offset as usize;
         let real_path = strip_path(path);
         let metadata = self.file_map.get(real_path).unwrap();
-
-        if offset > metadata.size {
+        let metadata_size = metadata.size as usize;
+        let mut chunks = metadata.chunks.clone();
+        chunks.sort_by(|(a, _), (b, _)| a.cmp(b));
+        let mut stack = self.stack.lock().unwrap();
+        let stack = stack
+            .entry(real_path.to_path_buf())
+            .or_insert(RefCell::new((vec![], 0, 0)));
+        let mut stack = stack.borrow_mut();
```

Comment:
This would be nicer as a block to make it clear when the lock on `self.stack` is released. If I understand this correctly, you keep the mutable borrow to the `RefCell` around, so you can release the lock on `self.stack`?

It's a cool solution, but would help if it was easier to read.

It would possibly also help to destructure these references if you can? Eg. `let (chunks, field2, current_read) = stack.borrow_mut();`

Reply:
I made a struct from the data, so it's more readable, and it let me get rid of the `RefCell` but I'm not quite sure what you mean by the first sentence.

Reply:
Eg.

```rust
let value = {
 let stack = self.stack.lock().unwrap(); // acquire lock
 
 // ... do something

 stack.pop()
}; // lock is released here, top of stack reference is in `value`
```

The above is just an example, but it's a pattern that should be helpful here (assuming no other changes).

Reply:
Oh yeah makes sense